src/main/java/org/opensearch/ad/rest/AbstractSearchAction.java

```diff
@@ -104,20 +137,20 @@ protected void onFailure(RestChannel channel, Exception e) {
         }
     }
 
-    protected RestResponseListener<SearchResponse> search(RestChannel channel) {
-        return new RestResponseListener<SearchResponse>(channel) {
-            @Override
-            public RestResponse buildResponse(SearchResponse response) throws Exception {
+    protected ExtensionRestResponse search(ExtensionRestRequest request, SearchResponse response) throws IOException {
+//        return new RestResponseListener<SearchResponse>(channel) {
```

Comment:
@dbwiddis need some suggestion here. Do we require a Transport API for `RestResponseListener` or CompletableFuture should be enough?

Reply:
@owaiskazi19 I'm not sure I understand the question here.  Is this a situation where we have a long-running request?

My initial thoughts are "wait some reasonable timeout for a fast request, if it's done send the search response, if not, send a response containing information to access the results later".

Reply:
Does that mean CompletableFuture should be enough to handle the `RestResponseListener` here?